core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -247,11 +247,39 @@ private ManifestFile copyManifest(ManifestFile manifest) {
    *
    * @param base table metadata to validate
    * @param startingSnapshotId id of the snapshot current at the start of the operation
-   * @param conflictDetectionFilter an expression used to find new conflicting data files
+   * @param conflictDetectionFilter a data expression used to find new conflicting data files
+   * @param caseSensitive whether expression evaluation should be case sensitive
+   */
+  protected void validateAddedFilesWithDataFilter(TableMetadata base, Long startingSnapshotId,
+                                                  Expression conflictDetectionFilter, boolean caseSensitive) {
+    validateAddedDataFiles(base, startingSnapshotId, conflictDetectionFilter, Expressions.alwaysTrue(), caseSensitive);
+  }
+
+  /**
+   * Validates that no files matching a filter have been added to the table since a starting snapshot.
+   *
+   * @param base table metadata to validate
+   * @param startingSnapshotId id of the snapshot current at the start of the operation
+   * @param conflictDetectionFilter a partition expression used to find new conflicting data files
+   * @param caseSensitive whether expression evaluation should be case sensitive
+   */
+  protected void validateAddedFilesWithPartFilter(TableMetadata base, Long startingSnapshotId,
```

Comment:
Rather than splitting the expression, I think you can write an alternative method that checks a `PartitionSet`.

Reply:
Done